app/components/market/submit/submit-market-application.component.ts

```diff
@@ -1,208 +1,171 @@
     constructor(
         public formBuilder: FormBuilder,
         private pythonRpcService: PythonRpcService,
         private route: ActivatedRoute,
         private router: Router,
-        private templateService: NcjTemplateService) { }
-
-    public ngOnInit() {
+        private templateService: NcjTemplateService) {
+             this.mode_state = "None";
+             this.form = new FormGroup({a: new FormControl()});
+    }
+    private ngOnInit() {
         this.route.params.subscribe((params) => {
             this.applicationId = params["applicationId"];
             this.actionId = params["actionId"];
-            this._updateTitle();
-            this._getTemplates();
-        });
-        this.route.queryParams.subscribe((params) => {
-            if (params.formParams) {
-                try {
-                    const value = JSON.parse(params.formParams);
-                    this._formValue = value;
-                    this.form.setValue(value);
-                } catch (e) {
-                    log.warn("Invalid form param. Not valid json.", params.formParams as any);
-                }
-            }
+            this.templateService.getTemplates(this.applicationId, this.actionId).subscribe((templates) => {
+                this.jobTemplate = templates.job;
+                this.poolTemplate = templates.pool;
+                this.title = `Run ${this.actionId} from ${this.applicationId}`;
+                this.createForms();
+            });
         });
     }
-
-    public createForms() {
-        let parameterKeys = Object.keys(this.jobTemplate.parameters);
+    private createForms() {
+        let jobParameters = [];
+        let poolParameters = [];
+        if (this.jobTemplate && this.jobTemplate.parameters){
+            jobParameters = Object.keys(this.jobTemplate.parameters);
+        }
+        if (this.poolTemplate && this.poolTemplate.parameters){
+            poolParameters = Object.keys(this.poolTemplate.parameters);
+        }
         let fg = {};
-        for (let key of parameterKeys) {
+
+        for (let key of jobParameters) {
             if ("defaultValue" in this.jobTemplate.parameters[key]) {
                 const defaultValue = String(this.jobTemplate.parameters[key].defaultValue);
                 fg[key] = new FormControl(defaultValue);
             } else {
                 fg[key] = new FormControl();
             }
         }
-        this.jobFormGroup = this.formBuilder.group(fg);
-
-        this.form = this.formBuilder.group({
-            job: this.jobFormGroup,
-            pool: this.pickedPool,
-        });
-        if (this._formValue) {
-            this.form.setValue(this._formValue);
+        for (let key of poolParameters) {
+            if ("defaultValue" in this.poolTemplate.parameters[key]) {
+                const defaultValue = String(this.poolTemplate.parameters[key].defaultValue);
+                fg[key] = new FormControl(defaultValue);
```

Comment:
This will be much better that merging all params togther and then trying to figure out which ones are belong to the pool and which ones to the job